blog/scripts/scripts.js

```diff
@@ -18,6 +18,21 @@ const PRODUCTION_DOMAINS = ['business.adobe.com'];
  * @param {Object} data additional data for RUM sample
  */
 export function sampleRUM(checkpoint, data = {}) {
+  sampleRUM.defer = sampleRUM.defer || [];
+  const defer = (fnname) => {
+    sampleRUM[fnname] = sampleRUM[fnname]
+      || ((...args) => sampleRUM.defer.push({ fnname, args }));
+  };
+  sampleRUM.drain = sampleRUM.drain
+    || ((dfnname, fn) => {
+      sampleRUM[dfnname] = fn;
+      sampleRUM.defer
+        .filter(({ fnname }) => dfnname === fnname)
+        .forEach(({ fnname, args }) => sampleRUM[fnname](...args));
+    });
```

Comment:
Nit: Personally, I think this would be more readable as a named function but I'm curious about what other devs think

Reply:
+1